src/lib/aws/common/construct.ts

```diff
@@ -48,8 +49,8 @@ import { CommonStackProps } from './types'
  *   };
  * };
  */
-export class CommonConstruct extends Construct {
-  props: CommonStackProps
+export class CommonConstruct<T extends CommonStackProps = CommonStackProps> extends Construct {
```

Comment:
allow subsclasses to define properties without needed to extend and override `props` with backward compatiblity

Reply:
We would need more meaningful names instead of using `T` to describe these

Reply:
these are generic and by convention they don't have much meaning expect conveying they are generic and restriction that apply to them. Good explanation https://stackoverflow.com/questions/68458146/why-do-we-use-the-letters-t-u-in-typescript-generics

Reply:
No generics yet please. It has been intentionally kept out

Reply:
https://gradientedge.atlassian.net/browse/MCA-608